pkg/security/probe/dentry_resolver.go

```diff
@@ -228,13 +231,16 @@ func (dr *DentryResolver) DelCacheEntry(mountID uint32, inode uint64) {
 			}
 			entries.Remove(key.Inode)
 
-			parent := path.(PathEntry).Parent
+			parent := path.(*PathEntry).Parent
 			if parent.Inode == 0 {
 				break
 			}
 
 			// Prepare next key
 			key = parent
+
+			// place the entry to the pool
+			dr.pathEntryPool.Put(path)
```

Comment:
I think this needs to be placed right below `entries.Remove(key.Inode)`. If a parent inode is null, we might not call `Put()` on a value that we removed from the cache.